src/h3lib/lib/faceijk.c

```diff
@@ -501,20 +501,28 @@ void _faceIjkToGeo(const FaceIJK* h, int res, GeoCoord* g) {
  *
  * @param h The FaceIJK address of the pentagonal cell.
  * @param res The H3 resolution of the cell.
+ * @param start The first topological vertex to return.
+ * @param length The number of topological vertexes to return.
  * @param g The spherical coordinates of the cell boundary.
  */
-void _faceIjkPentToGeoBoundary(const FaceIJK* h, int res, GeoBoundary* g) {
+void _faceIjkPentToGeoBoundary(const FaceIJK* h, int res, int start, int length,
+                               GeoBoundary* g) {
     int adjRes = res;
     FaceIJK centerIJK = *h;
     FaceIJK fijkVerts[NUM_PENT_VERTS];
     _faceIjkPentToVerts(&centerIJK, &adjRes, fijkVerts);
 
+    // If we're returning the entire loop, we need one more iteration in case
+    // of a distortion vertex on the last edge
+    int additionalIteration = length == NUM_PENT_VERTS ? 1 : 0;
```

Comment:
This would only be returned for the full boundary? Seems like it should be addressable either with the first or last standard vertex?

Reply:
This only applies to the full boundary. If I'm taking the subset `vertex 4, vertex 0`, then I'll already be considering the possible distortion vertex between 4 and 0, because I consider `4 - 0`. But if I'm taking `vertex 0, vertex 4`, then I wouldn't consider the final `4 - 0` edge, because I only consider `0 - 1 - 2 - 3 - 4`. It's only when I'm closing the loop that I need an additional iteration to check the last edge, since it's not part of my fencepost sequence.